### ISSUE TITLE
Simplify access to library documentation

### ISSUE THREAD

**Issue by igaztanaga** [boostorg/website-v2#1008]
For users, access to the documentation of a library is not simple enough. See [this comment](https://www.reddit.com/r/cpp/comments/1bdz971/comment/kuq916h/) from Reddit:

> I need to go
> latest release
> documentation
> blob of text with me having to find an inline link in section 4, libraries.
> unordered
> automatic redirect failed
> 
> or
> 
> learn
> intro? no
> getting started? no
> explore the content? no
> literally any of these links is useless. completely.
> 
> find libraries on top
> unordered, but in a hard to read list
> documentation
> automatic redirect failed

A suggestion to fix this issue would be to reserve the top section of the main page to the most popular use cases, like downloads and documentation, and put "why use boost", "mission" and other information that will be read just by new users in a less visible part (.e.g. lower in the page or in a different page).

The documentation link can be put nearly at the same level as the current "Download the latest release" that says "Documentation for the latest release" that would redirect to something like "https://www.boost.io/doc/libs/boost_1_84_0/libs/libraries.htm". Alternativelly, we can have a drop-down or similar menu in the front page instead of the "Documentation for the latest release" button where a user selects a library (e.g. Unordered) and the selection redirects to the [documentation of the latest release of Unordered](https://www.boost.io/doc/libs/boost_1_84_0/libs/unordered/doc/html/unordered.html).

**Comment by alandefreitas** [comment]
> Feedback, which has always annoyed me on the old page as well: As a user, I don't care about the Boost mission, I want
> the download

That's available on the home page.

> the documentation

That's on the top already as "Learn". Maybe that's not very intuitive.

> the repo

The link is available on the home page but I think incentivizing people to download the release instead of going to the super-project repo is intentional.

> To get to the (randomly picked) Boost.Unordered documentation for the latest version, I need to go
> latest release, documentation, blob of text with me having to find an inline link in section 4, libraries., unordered, automatic redirect failed

So, he didn't really "need to" do that. He could go "Libraries" -> "Unordered" -> "Documentation". Whether the website is making this less intuitive than the alternative is something to discuss. My guess is he was already at the download page and tried to get to unordered from there.

He mentions that later in

> find libraries on top -> unordered, but in a hard to read list

I don't know if it's easy to make that list easier to read because the list is just too long. It would be nice to have a small button to the docs in each card but I think this is already proposed in another issue.

> For the repo: latest release -> source code -> code tab

I didn't know where the link to the repo was. I just went ctrl+f "github" and found it in the home page. Since it's there AND also in the latest release page (where he didn't  _need to_ go), I think that's fine.

At last, the way he ends his comment is catastrophizing too much in my opinion.

**Comment by igaztanaga** [comment]
Yes, we don't need to take all comments seriously. Just that below the "Download the Latest Release" button we could put some shortcuts to the most used features like Documentation for the latest version or the "https://github.com/boostorg" link where all Boost code is hosted.

**Comment by vinniefalco** [comment]
I am very much in favor of using site logs to analyze where people go the most and optimizing the interface for that
